src/SaveObject.ts

```diff
@@ -110,115 +117,103 @@ class BitburnerSaveObject {
 
     if (Player.gang) this.AllGangsSave = JSON.stringify(AllGangs);
 
-    const saveString = btoa(unescape(encodeURIComponent(JSON.stringify(this))));
-    return saveString;
+    return await encodeJsonSaveString(JSON.stringify(this), Settings.CompressSaveData);
   }
 
-  saveGame(emitToastEvent = true): Promise<void> {
+  async saveGame(emitToastEvent = true): Promise<void> {
     const savedOn = new Date().getTime();
     Player.lastSave = savedOn;
-    const saveString = this.getSaveString();
-    return new Promise((resolve, reject) => {
-      save(saveString)
-        .then(() => {
-          const saveData: SaveData = {
-            playerIdentifier: Player.identifier,
-            fileName: this.getSaveFileName(),
-            save: saveString,
-            savedOn,
-          };
-          pushGameSaved(saveData);
-
-          if (emitToastEvent) {
-            SnackbarEvents.emit("Game Saved!", ToastVariant.INFO, 2000);
-          }
-          return resolve();
-        })
-        .catch((err) => {
-          console.error(err);
-          return reject();
-        });
-    });
+    const saveData = await this.getSaveData();
+    await save(saveData);
+    const electronGameData: ElectronGameData = {
+      playerIdentifier: Player.identifier,
+      fileName: this.getSaveFileName(),
+      save: saveData,
+      savedOn,
+    };
+    pushGameSaved(electronGameData);
+
+    if (emitToastEvent) {
+      SnackbarEvents.emit("Game Saved!", ToastVariant.INFO, 2000);
+    }
   }
 
   getSaveFileName(isRecovery = false): string {
     // Save file name is based on current timestamp and BitNode
     const epochTime = Math.round(Date.now() / 1000);
     const bn = Player.bitNodeN;
-    let filename = `bitburnerSave_${epochTime}_BN${bn}x${Player.sourceFileLvl(bn) + 1}.json`;
+    /**
+     * Old behavior: save file uses .json extension. Save data is json save string in base64 format.
+     *
+     * New behavior:
+     * - Enable compression: save file uses .json.gz extension. Save data is json save string in binary format.
+     * - Disable compression: save file uses .save extension. Save data is json save string in base64 format.
```

Comment:
```suggestion
     * - Compression available: save file uses .json.gz extension. Save data is gzipped json save string in binary format.
     * - Compression unavailable: save file uses .json extension. Save data is json save string in base64 format.
```

Same note about "comments document the current code."